gef.py

```diff
@@ -8159,14 +8160,23 @@ def context_trace(self):
         nb_backtrace = self.get_setting("nb_lines_backtrace")
         if nb_backtrace <= 0:
             return
-        orig_frame = current_frame = gdb.selected_frame()
-        i = 0
 
         # backward compat for gdb (gdb < 7.10)
         if not hasattr(gdb, "FrameDecorator"):
             gdb.execute("backtrace {:d}".format(nb_backtrace))
             return
 
+        orig_frame = gdb.selected_frame()
+        current_frame = gdb.newest_frame()
+        frames = [current_frame]
+        while current_frame != gdb.selected_frame():
```

Comment:
From the other PR we discovered that `selected_frame` builds a new object every time it is called, so it would create a new one for every frame in the stack.
```suggestion
        selected_frame = gdb.selected_frame()
        while current_frame != selected_frame:
```

Reply:
Comparison operators are overloaded for gdb.Frame so it works correct.

Reply:
Thank you for suggestion. We can even use `orig_frame`.

Reply:
Right! Use that.

Reply:
Fixed.